src/controllers/version/docs.rs

```diff
@@ -0,0 +1,107 @@
+//! Endpoint for triggering a docs.rs rebuild
+
+use super::CrateVersionPath;
+use crate::app::AppState;
+use crate::auth::AuthCheck;
+use crate::util::errors::{AppResult, server_error};
+use crate::worker::jobs;
+use axum::response::{IntoResponse as _, Response};
+use crates_io_worker::BackgroundJob as _;
+use http::StatusCode;
+use http::request::Parts;
+
+/// Trigger a rebuild for the crate documentation on docs.rs.
+#[utoipa::path(
+    post,
+    path = "/api/v1/crates/{name}/{version}/rebuild_docs",
+    params(CrateVersionPath),
+    security(
+        ("cookie" = []),
+    ),
+    tag = "versions",
+    responses((status = 201, description = "Successful Response")),
+)]
+pub async fn rebuild_version_docs(
+    app: AppState,
+    path: CrateVersionPath,
+    req: Parts,
+) -> AppResult<Response> {
+    let mut conn = app.db_write().await?;
+    AuthCheck::only_cookie().check(&req, &mut conn).await?;
```

Comment:
I just realized on thing is missing here: 

I need to validate if the logged in user is a maintainer (is that how it's called in here?) of the crate

Reply:
It's called `Rights`. You should be able to find the relevant code in the update/delete endpoint, for example: https://github.com/rust-lang/crates.io/blob/c00a84db6992ec44d97af022674f2e0fe176c86e/src/controllers/krate/delete.rs#L69-L82.

Reply:
added the permission check